### ISSUE TITLE
[ios_platform_images] Platform tests fail in CI with Xcode 15 and iOS 17

### ISSUE THREAD

**Issue by vashworth** [flutter/flutter#136827]
`xcode analyze` step fails with the following error:

```
While building module 'ios_platform_images' imported from /Volumes/Work/s/w/ir/x/w/packages/packages/ios_platform_images/example/ios/Runner/GeneratedPluginRegistrant.m:16:
In file included from <module-includes>:1:
/Users/chrome-bot/Library/Developer/Xcode/DerivedData/Runner-ebfdwekzmldmdefbogamgoyvamxm/Build/Products/Debug-iphonesimulator/ios_platform_images/ios_platform_images.framework/Headers/ios_platform_images-umbrella.h:20:1: error: umbrella header for module 'ios_platform_images' does not include header 'ios_platform_images-Swift.h' [-Werror,-Wincomplete-umbrella]
1 error generated.
/Volumes/Work/s/w/ir/x/w/packages/packages/ios_platform_images/example/ios/Runner/GeneratedPluginRegistrant.m:16:9: fatal error: could not build module 'ios_platform_images'
#import <ios_platform_images/IosPlatformImagesPlugin.h>
 ~~~~~~~^
2 errors generated.
```

https://ci.chromium.org/ui/p/flutter/builders/try/Mac_arm64%20ios_platform_tests_shard_3%20master/7590/overview
https://ci.chromium.org/ui/p/flutter/builders/try/Mac_arm64%20ios_platform_tests_shard_3%20master/7567/overview

I also tried temporarily skipping the `xcode analyze` step, but the `native test` step also fails

```
While building module 'ios_platform_images' imported from /Volumes/Work/s/w/ir/x/w/packages/packages/ios_platform_images/example/ios/Runner/GeneratedPluginRegistrant.m:16:
In file included from <module-includes>:1:
/Users/chrome-bot/Library/Developer/Xcode/DerivedData/Runner-ebfdwekzmldmdefbogamgoyvamxm/Build/Products/Debug-iphonesimulator/ios_platform_images/ios_platform_images.framework/Headers/ios_platform_images-umbrella.h:20:1: error: umbrella header for module 'ios_platform_images' does not include header 'ios_platform_images-Swift.h' [-Werror,-Wincomplete-umbrella]
1 error generated.
/Volumes/Work/s/w/ir/x/w/packages/packages/ios_platform_images/example/ios/Runner/GeneratedPluginRegistrant.m:16:9: fatal error: could not build module 'ios_platform_images'
#import <ios_platform_images/IosPlatformImagesPlugin.h>
 ~~~~~~~^
2 errors generated.

...

Umbrella header for module 'ios_platform_images' does not include header 'ios_platform_images-Swift.h'
	Could not build module 'ios_platform_images'
	Testing cancelled because the build failed.
```
https://ci.chromium.org/ui/p/flutter/builders/try/Mac_arm64%20ios_platform_tests_shard_3%20master/7593/overview

Strangely the `build examples` step passes fine.

Also, I was unable to recreate this locally. Both the analyze and test work for me locally.

Also, doesn't seem to happen every time: https://luci-milo.appspot.com/ui/p/flutter/builders/try/Mac_arm64%20ios_platform_tests_shard_3%20master/7545/overview

Not sure if this is a flake or legit issue

**Comment by stuartmorgan** [comment]
I can't repro locally either. I was wondering if it was due to cached build output from the PR that did the Swift migration, where some earlier run might have left bridging header that shouldn't be there, but I've verified that the generated header is present locally when I build. There's just no error about it not being included in the umbrella.

It looks like the umbrella is generated by Cocoapods in this scenario, so it could be a bug there, but I don't think the generated -Swift.h bridging header is *supposed* to be in the umbrella based on what I can find (which isn't anything official so far).
